### PR TITLE
Added ACM for SLL and Discovery Mode Configurations

### LINKED ISSUE #1591: Add PeerForwarderConfiguration class
Create PeerForwarderConfiguration class and embed it in to DataPrepperCOnfiguration.

### PR DESCRIPTION
**Pull request by asifsmohammed**
Signed-off-by: Asif Sohail Mohammed <nsifmoh@amazon.com>

### Description
- Added Discovery Mode configuration option with Static, DNS, Cloud Map Discover modes
- Added ACM support for SSL
 
### Issues Resolved
Resolves #1591 
 
### Check List
- [x] New functionality includes testing.
- [ ] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-core/src/test/java/com/amazon/dataprepper/peerforwarder/PeerForwarderConfigurationTest.java [hunk 1]
```diff
@@ -75,9 +89,17 @@ void testValidPeerForwarderConfig() throws IOException {
...
+            INVALID_PEER_FORWARDER_WITH_ACM_WITHOUT_REGION_CONFIG_FILE,
+            INVALID_PEER_FORWARDER_WITH_CLOUD_MAP_WITHOUT_SERVICE_NAME_CONFIG_FILE,
+            INVALID_PEER_FORWARDER_WITH_CLOUD_MAP_WITHOUT_NAMESPACE_NAME_CONFIG_FILE,
+            INVALID_PEER_FORWARDER_WITH_CLOUD_MAP_WITHOUT_REGION_CONFIG_FILE,
+            INVALID_PEER_FORWARDER_WITH_DNS_WITHOUT_DOMAIN_NAME_CONFIG_FILE
     })
-    void invalid_InvalidPeerForwarderConfig_test(final String filePath) {
+    void invalid_InvalidPeerForwarderConfig_test(final String filePath) throws IOException {
+//        PeerForwarderConfiguration peerForwarderConfiguration = makeConfig(filePath);
+//        System.out.println(peerForwarderConfiguration.isSsl());
```
Comment (chenqi0805) [inline data-prepper-core/src/test/java/com/amazon/dataprepper/peerforwarder/PeerForwarderConfigurationTest.java:102]: Remove commented out code.

File: data-prepper-core/src/test/java/com/amazon/dataprepper/peerforwarder/discovery/AwsCloudMapPeerListProvider_CreateTest.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class AwsCloudMapPeerListProvider_CreateTest {
```
Comment (chenqi0805) [inline data-prepper-core/src/test/java/com/amazon/dataprepper/peerforwarder/discovery/AwsCloudMapPeerListProvider_CreateTest.java:28]: We should change this name pattern into `AwsCloudMapPeerListProviderCreateTest`

File: data-prepper-core/src/test/java/com/amazon/dataprepper/peerforwarder/discovery/DnsPeerListProvider_CreateTest.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class DnsPeerListProvider_CreateTest {
```
Comment (chenqi0805) [inline data-prepper-core/src/test/java/com/amazon/dataprepper/peerforwarder/discovery/DnsPeerListProvider_CreateTest.java:30]: Same as above

File: data-prepper-core/src/test/java/com/amazon/dataprepper/peerforwarder/discovery/StaticPeerListProvider_CreateTest.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class StaticPeerListProvider_CreateTest {
```
Comment (chenqi0805) [inline data-prepper-core/src/test/java/com/amazon/dataprepper/peerforwarder/discovery/StaticPeerListProvider_CreateTest.java:23]: Same as above

**Review by chenqi0805 — APPROVED** [review]

**Review by dlvenable — APPROVED** [review]
This looks good. Thanks!

One high-level thought: There might be a good opportunity to use the plugin framework for discovery modes. This would make for a nested configuration and allow the addition of other discovery modes easily. I think it is fair to put that beyond the scope of this PR though.

**Comment by asifsmohammed** [comment]
> This looks good. Thanks!
> 
> One high-level thought: There might be a good opportunity to use the plugin framework for discovery modes. This would make for a nested configuration and allow the addition of other discovery modes easily. I think it is fair to put that beyond the scope of this PR though.

Does it mean using pluginModel for Discover Mode?

**Comment by dlvenable** [comment]
Yes. We could use the pluginModel for the discovery mode. This way, other discovery options can be provided without changing Data Prepper core.

**Pull request merged**